Add kubeconform based manifest validation

naoki9911 opened the pull request:
This PR adds validation for manifests with kubeconform's validator.

## Changes
- **Breaking Change**: Manifests for policies and resources are validated against schema by default.
  - Option `--validate-resource-manifests` is added to toggle manifest validation. (enabled by default)
- `SchemaLocation` field is added to specify the path to schemas for CRD.

naoki9911 committed:
Bump go version to 1.24

naoki9911 committed:
Add resource manifest validator

naoki9911 committed:
Fix resources.yaml to pass test

naoki9911 committed:
Support CRD in resource manifest validation

naoki9911 committed:
Validate policies when resource manifest validation enabled

naoki9911 committed:
Add tests for CRD and disabled validation

naoki9911 committed:
README.md: Add 'Resource Manifest Validation'

naoki9911 committed:
Bump golanci-lint version to v1.64

naoki9911 committed:
CREDITS: Updated

superbrothers commented on internal/tester/tester.go:
```diff
@@ -104,8 +111,41 @@ func runEach(cfg CmdConfig, manifestPath string) testResultSummary {
+			SkipTLS:              false,
+			SkipKinds:            map[string]struct{}{},
+			RejectKinds:          map[string]struct{}{},
+			KubernetesVersion:    "1.32.1", // ensure matching a version with validation.go and mutation.go
```
I'm afraid I'll forget to update easily. Is there any good solution?

naoki9911 replied:
Fixed in https://github.com/pfnet/kaptest/pull/11/commits/b0680357c2eaade96c2d6fcba3d86ed1acb8c06e .
Supported kubernetes version is now retrieved from dependent k8s.io/api module's version.

naoki9911 committed:
Use k8s.io/api version as supported kubernetes version

This patch embeds go.mod and retrieves k8s.io/api's version
as supported kubernetes version.
debug.ReadBuildInfo also provides dependent module's info,
but they requires built binary and cannnot use it at tests.
